Feature: clickable folders in table of contents

sjorek opened the pull request:
Well, what else should I write ? … it was too annoying, so I implemented clickable folders in the table of contents. Everything works as expected - including keyboard-navigation. (I didn't touch it anyway). I hope I didn't miss anything … well I did not test empty folders as I was not able to reproduce such a combination (empty but visible).

This should also fix: https://github.com/nevir/groc/pull/80 …

So I stay calm and look forward to a successful merge of this pull-request.

Stephan :zzz:

sjorek committed:
made folders clickable in toc

sjorek committed:
bugfixed broken discloser for the sake of clickable folders

sjorek commented:
… yep, I sadly broke the discloser … :boom: … and bugfixed it. Sorry, for not testing this nifty thingy thoroughly enough …

It works perfect for me. :ok_hand:

kmdavis commented:
Sounds like a good idea. I should have time on monday to look at (and test) this and the other new pull requests.

sjorek commented:
Yeah, cool. :sunglasses: Keeping the amount of different branches low, makes the process much more maintainable for me. 

Sadly I'm only partially available on monday. Nevertheless, as my timezone is probably some (6-8 ?) hours ahead compared to yours, there might be a good chance that I can react (in time) at your daytime if anything is wrong or needs clarification. And actually I'm absolutely not in a hurry.

If you find time to look at [**my fork** of _groc_](https://github.com/sjorek/groc/), you might notice that I already implemented more features, like:
- enhanced code-folding and comment-embedding …
  - … in https://github.com/sjorek/groc/blob/master/lib/utils.coffee#L241
  - … which looks like http://sjorek.github.io/groc/utils.html at the line **“Unfold this code ->”**
- support for coffee-script's
  [block-comments](https://github.com/sjorek/groc/blob/master/lib/languages.coffee#L50)
  (as promised)
- support for [literate coffee-script _(*.litcoffee)_](http://coffeescript.org/#literate):
  - A feature I missed very much. I used it to
    [document upcoming changes](http://sjorek.github.io/groc/languages/namespace.html)
  - in https://github.com/sjorek/groc/blob/master/lib/languages.coffee#L73
  - and https://github.com/sjorek/groc/blob/master/lib/utils.coffee#L108
- support for [JSON _(*.json)_](http://json.org), because upcoming features
  [use them](http://sjorek.github.io/groc/languages/namespace/javascript_dom.html) and
  I wanted them to be “documented” … 
  - see https://github.com/sjorek/groc/blob/master/lib/languages.coffee#L163
  - I know there are no comments inside JSON-files, but at least the source-code can be included in groc's output
  - There is [JSON-Shema](http://json-schema.org) as a source inspiration on how to get documented output
    without corrupting the JSON-File itself.  The namespace-implementation mentioned below, could be
    extended to get support for this feature (but I'm already satisfied with the current state)
- additional enhancements and additions for various `@doctags` eg.:
  - support for `@author`
    (done, tested)
  - support for `@see`
    (done, tested)
  - support for `@fires`
    (done, not tested)
  - added the missing humanize in `@return` and `@type`
    (done, tested)
  - refactored most of the @doctag-implementation, to be able to move
    different language-dependend aspects into separate implementations
    (work in progress, 99% done, I need this for my own projects)
  - language dependent [hyperlinked namespaces and types](http://sjorek.github.io/groc/utils/doctag_helpers.html)
    used in `@param`-, `@return`\- and `@type`-doctags.
    (work in progress, 70% done, I need this for my own projects)
  - and surely more …

I promise to offer all changes as pull requests in separate, small and clean branches, to keep your overhead as low as possible (and of course mine too). And I'll do it step by step … not everything at once, as we don't want to break anything. If you want to, you could specify the order you'd like to process them, so I can focus on the changes from higher to lower priorities. Or tell me if there is something you absolutely don't ever want to include. :speak_no_evil: :hear_no_evil: :see_no_evil:

**There is one (and a half :smirk:) question bothering me**: Do you see any chance to raise the node-version requirement at least to node ≥ 0.8.0, if we incorporate some or all of the features I mentioned above ? (I think bumping groc's version to 0.7.0 and using a separate branch might be a good idea then …)

Cheerio,
Stephan
